### PR TITLE
Sponsor Form UI update and form validations

### PR DESCRIPTION
**Pull request by claudianguyen**
- Updated the Sponsor Form using Formik and Yup for form validations 
- Used React Bootrap Form for styling
- Added Street address, zip, state, country fields -> Will need to figure out how to merge a address object when we send to backend
- Added phone_number to Sponsor Table

### CODE REVIEW

File: frontend/jsx/Main.jsx [hunk 1]
```diff
@@ -62,20 +70,20 @@ class Main extends Component {
...
 
-    /**
-     * Handler for sponsor info changes.
+     /**
+     * Handler for the Sponsor Form 'NEXT' button click. 
+     * Takes in 'values' object which contains an object of fields to values on the form.
      */
-    sponsorInfoChangeHandler(field) {
+    handleSponsorFormClick(values) {
+        console.log("values from main.jsx:", values )
```
Comment (ShahilZ) [inline frontend/jsx/Main.jsx:78]: Can remove the `console.log()` in line 78

File: frontend/jsx/Main.jsx [hunk 2]
```diff
@@ -62,20 +70,20 @@ class Main extends Component {
...
-                sponsor: {
-                    ...prevState.sponsor,
-                    [field]: newValue
-                },
-            }));
-        }
+        this.setState({ sponsor: values.sponsor });
+
+        console.log("state from main.jsx:", this.state.sponsor )
+        //console.log("hellllllo");
```
Comment (ShahilZ) [inline frontend/jsx/Main.jsx:84]: Same here: `hellllllllllllllllllllllllllllllllllo`

File: frontend/jsx/common/sponsorship/SponsorForm.jsx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+                                            type="text"
+                                            name="sponsor.lastName"
+                                            value={values.sponsor.lastName}
+                                            onChange={handleChange}
+                                            isInvalid={getIn(touched, "sponsor.lastName") && getIn(errors, "sponsor.lastName")}
+                                            isValid={getIn(touched, "sponsor.lastName") && !getIn(errors, "sponsor.lastName")}
+                                            placeholder={i18n.t("sponsor_info:form_placeholder_last_name")}
+                                        />
+                                        <Form.Control.Feedback type="invalid">
+                                            {`${i18n.t("sponsor_info:form_error_last_name")}`}
```
Comment (ShahilZ) [inline frontend/jsx/common/sponsorship/SponsorForm.jsx:99]: Line 99: `i18n.t(...)`

File: frontend/jsx/common/sponsorship/SponsorForm.jsx [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+                                        </Form.Control.Feedback>
+                                        <Form.Control.Feedback type="valid" />
+                                    </Form.Group>
+                                </Form.Row>
+                                <Form.Row>
+                                    <Form.Group as={Col} controlId="email">
+                                        <Form.Label>{`${i18n.t("sponsor_info:form_email")}`}</Form.Label>
```
Comment (ShahilZ) [inline frontend/jsx/common/sponsorship/SponsorForm.jsx:106]: Same here

File: frontend/jsx/common/sponsorship/SponsorForm.jsx [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+                                            type="text" 
+                                            name="sponsor.email"
+                                            value={values.sponsor.email}
+                                            onChange={handleChange}
+                                            isInvalid={getIn(touched, "sponsor.email") && getIn(errors, "sponsor.email")}
+                                            isValid={getIn(touched, "sponsor.email") && !getIn(errors, "sponsor.email")}
+                                            placeholder={i18n.t("sponsor_info:form_placeholder_email")}
+                                        />
+                                        <Form.Control.Feedback type="invalid">
+                                           {`${i18n.t("sponsor_info:form_error_email")}`}
```
Comment (ShahilZ) [inline frontend/jsx/common/sponsorship/SponsorForm.jsx:117]: Here too

File: frontend/jsx/common/sponsorship/SponsorForm.jsx [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+                                        />
+                                    </Form.Group>
+                                </Form.Row>
+                                 <Form.Group>
+                                    <Form.Check
+                                        label={i18n.t("sponsor_info:form_newsletters_subscription_checkbox")}
+                                        inline
+                                        custom
+                                        name="sponsor.subscribed"
+                                        id="subsribed"
```
Comment (ShahilZ) [inline frontend/jsx/common/sponsorship/SponsorForm.jsx:152]: `subscribed`

File: frontend/jsx/common/sponsorship/SponsorForm.jsx [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+                                        checked={values.sponsor.subscribed}
+                                        type="checkbox"
+                                        onChange={() => setFieldValue("subscribed", !values.sponsor.subscribed)}
+                                    />
+                                </Form.Group>
+                            </Col>
+                            <Col xl={6}>
+                                <div className="mb-3">
+                                    <h4>{`${i18n.t("sponsor_info:form_header_address")}`}</h4>
```
Comment (ShahilZ) [inline frontend/jsx/common/sponsorship/SponsorForm.jsx:161]: No need for jsx interpolation

File: frontend/jsx/common/sponsorship/SponsorForm.jsx [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+                                </div>
+                                <Form.Row>
+                                    <Form.Group as={Col} controlId="streetAddress">
+                                        <Form.Label>Street Address</Form.Label>
+                                        <Form.Control 
+                                            placeholder="Street name, number, P.O box, Apt #"
```
Comment (ShahilZ) [inline frontend/jsx/common/sponsorship/SponsorForm.jsx:167]: Does this need to be i18n-ed?

File: frontend/package.json [hunk 9]
```diff
@@ -1,63 +1,65 @@
-{
```
Comment (ShahilZ) [inline frontend/package.json:1]: Verify no changes besides your additions please

**Review by ShahilZ — COMMENTED** [review]
Looks good, just few syntax things + i18n. Please verify that `package.json` is not too different

File: frontend/jsx/Main.jsx (hunk 1 above)
Comment (claudianguyen) [reply frontend/jsx/Main.jsx:78]: will do after i put donation changes

**Pull request merged**